src/components/AssetManagementPanel/AssetManagementContainer.tsx

```diff
@@ -0,0 +1,100 @@
+import React, { FunctionComponent, useState, useEffect } from "react";
+import { WrappedDocument } from "@govtechsg/open-attestation";
+import { TitleEscrow } from "@govtechsg/token-registry/types/TitleEscrow";
+import { AssetManagementTags } from "./AssetManagementTags";
+import { getDocumentId, getTokenRegistryAddress } from "../../common/utils/document";
+import { useTitleEscrowContract } from "../../common/hooks/useTitleEscrowContract";
+import { useProviderContext } from "../../common/contexts/provider";
+import { useContractFunctionHook } from "@govtechsg/ethers-contract-hook";
+import { AssetManagementForm } from "./AssetManagementForm";
+
+export enum AssetManagementActions {
+  None = "None",
+  TransferHolder = "TransferHolder",
+  EndorseBeneficiary = "EndorseBeneficiary",
+  Surrender = "Surrender",
+}
+
+interface AssetManagementApplication {
+  tokenId: string;
+  tokenRegistryAddress: string;
+  titleEscrow: TitleEscrow;
+}
+
+export const AssetManagementApplication: FunctionComponent<AssetManagementApplication> = ({
+  tokenId,
+  tokenRegistryAddress,
+  titleEscrow,
+}) => {
+  const [assetManagementAction, setAssetManagementAction] = useState(AssetManagementActions.None);
+  const { isUpgraded, upgradeProvider, account } = useProviderContext();
+  const { call: getHolder, value: holder } = useContractFunctionHook(titleEscrow, "holder");
+  const { call: getBeneficiary, value: beneficiary } = useContractFunctionHook(titleEscrow, "beneficiary");
+  const { call: getApprovedTransferTarget, value: approvedTransferTarget } = useContractFunctionHook(
+    titleEscrow,
+    "approvedTransferTarget"
+  );
+  const { send: sendSurrender, state: surrenderingState } = useContractFunctionHook(titleEscrow, "transferTo");
+
+  // const isHolder = account === holder;
```

Comment:
Can remove if we are not doing it at this level

Reply:
can i keep the surrenderState outside? passing it into AssetManagement form so storybook can render `PENDING_CONFIRMATION` state.